pgrx/src/fcinfo.rs

```diff
@@ -332,30 +335,30 @@ unsafe fn direct_function_call_as_datum_internal(
     args: &[Option<pg_sys::Datum>],
 ) -> Option<pg_sys::Datum> {
     let nargs: i16 = args.len().try_into().expect("too many args passed to function");
-    let fcinfo_ptr = pg_sys::palloc(
+    let fcinfo = pg_sys::palloc0(
         std::mem::size_of::<pg_sys::FunctionCallInfoBaseData>()
             + std::mem::size_of::<pg_sys::NullableDatum>() * args.len(),
     )
     .cast::<pg_sys::FunctionCallInfoBaseData>();
 
-    let fcinfo = fcinfo_ptr.as_mut().unwrap_unchecked();
```

Comment:
This is a dubious-soundness action due to later interactions with the args slice being somewhat incompatible with Stacked Borrows.